### DISCUSSION TITLE
Why does TopoStats hang once it look like its finished processing?

### DISCUSSION THREAD

**Discussion by Max-Gamill** [AFM-SPM/TopoStats#710] [Q&A]
There seems to be an issue with the multiprocessing workers and them returning to the pool causing the running of TopoStats to hang with the inclusion of the Topoly package.

The solution to this is to kill the hanging python process and it should continue to run fine and output the `csv` and `json` files.

This can be done on MacOS via identifying the python process in the doc, click and hold, and pressing "Quit".
On Windows / Linux machines you can terminate the processes through Task Manager and [kill terminal command](https://www.cyberciti.biz/faq/kill-process-in-linux-or-terminate-a-process-in-unix-or-linux-systems/) respectively.

You should see a new python process appear and TopoStats giving the completion message.

**Comment by ns-rse** [comment]
You can also use `killall python` on GNU/Linux and OSX systems from a terminal (above linked solutions using `kill` also work on OSX). If that doesn't work then try again using the `KILL` signal  which is `9` (default is `TERM`) with `killall -s 9 python`.

These and the `kill` solution linked above also work on OSX from a terminal (as OSX's pedigree stems from the UNIX BSD).
